src/emojis/EmojiWrapper.tsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright 2023 Kapeta Inc.
+ * SPDX-License-Identifier: MIT
+ */
+
+import React from 'react';
+import { Box, BoxProps } from '@mui/material';
+
+export type EmojiSize = 'small' | 'medium' | 'large' | number;
+
+type Dimension = {
+    width: number;
+    height: number;
+};
+
+const getEmojiSizeInPx = (size: EmojiSize): number => {
+    switch (size) {
+        case 'small':
+            return 20;
+        case 'medium':
+            return 24;
+        case 'large':
+            return 35;
+        default:
+            return size;
+    }
+};
+
+const getEmojiSize = (svgDimension: Dimension, size: EmojiSize): Dimension => {
+    const sizeInPx = getEmojiSizeInPx(size);
+    const ratio = svgDimension.width / svgDimension.height;
+
+    let width, height;
+
+    if (svgDimension.width > svgDimension.height) {
+        width = sizeInPx;
+        height = sizeInPx / ratio;
+    } else {
+        width = sizeInPx * ratio;
+        height = sizeInPx;
+    }
+
+    // Ensure the dimensions do not exceed the max size
+    width = Math.min(width, sizeInPx);
+    height = Math.min(height, sizeInPx);
```

Comment:
this doesn't seem right, it ignores the original ratio. Don't we risk it getting squashed?

Reply:
It uses the ratio in the if-else above. But I pushed something new that should be better

Reply:
Much more readable now, thanks